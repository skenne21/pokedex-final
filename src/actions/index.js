export const setTypes = (types) => ({
 type: 'SET_TYPES',
 types
})
