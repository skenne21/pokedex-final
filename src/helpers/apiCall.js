export const getTypes =  async () => {
  try {
    const response = await fetch('http://localhost:3001/types');
    const apiData = await response.json();
    return apiData;
  } catch (error) {
    throw error
  }
}

export const getPokemon = async (ids) => {
  const pokemonPromises = ids.map(async pokemon => {
    const apiData = await fetchPokemon(pokemon)
    const indivdualPokemon = {
      name: apiData.name,
      weight: apiData.weight,
      image: apiData.sprites.back_default
    }
    return indivdualPokemon;
  })  
  return Promise.all(pokemonPromises)
}

export const fetchPokemon = async (pokemon) => {
  try {
    const response = await fetch(`http://localhost:3001/pokemon/${pokemon}`);
    const apiData = await response.json();
    return apiData;
  } catch (error) {
    throw error
  }  
}


