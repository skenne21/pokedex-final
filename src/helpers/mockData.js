export const types = [
  {
    id:"1",
    name: "normal",
    pokemon: [
      '16',
      '17'
    ]
  }, {
    id: "2",
    name: "fighting",
    pokemon: [
      "56",
      "57",
      "66",
    ]
  }
]