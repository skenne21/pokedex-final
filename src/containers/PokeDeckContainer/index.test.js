import React from 'react';
import { shallow } from 'enzyme';
import { PokeDeckContainer } from './index';
import { mapStateToProps, mapDispatchToProps } from './index';
import * as actions from '../../actions';
import * as apicalls from '../../helpers/apiCall';
import * as mocks from '../../helpers/mockData';

describe('PokeDeckContainer', () => {
  let wrapper, setTypes, types;

  beforeEach(() => {
    setTypes = jest.fn();
    types = mocks.types;
    wrapper = shallow(
      <PokeDeckContainer
        setTypes={setTypes}
        types={types}
      />
    )
  });

  it('Should match the snap shot if there is not any types', () => {
    wrapper = shallow(
      <PokeDeckContainer
        setTypes={setTypes}
        types={[]}
      />
    )
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match the snapshot if the types have length', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    let expectedState, mappedState;
    
    beforeEach(() => {
      expectedState = { types:mocks.types };
      mappedState = { types:mocks.types };
    });
    
  });
});