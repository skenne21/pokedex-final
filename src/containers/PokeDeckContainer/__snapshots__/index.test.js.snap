// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeDeckContainer Should match the snap shot if there is not any types 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokeDeckContainer
    setTypes={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
      }
    }
    types={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src="../../../public/pikachu.gif"
        />,
        <Lists
          pokemons={undefined}
        />,
      ],
      "className": "PokeDeckContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "../../../public/pikachu.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "pokemons": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src="../../../public/pikachu.gif"
          />,
          <Lists
            pokemons={undefined}
          />,
        ],
        "className": "PokeDeckContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "../../../public/pikachu.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "pokemons": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PokeDeckContainer Should match the snapshot if the types have length 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokeDeckContainer
    setTypes={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
      }
    }
    types={
      Array [
        Object {
          "id": "1",
          "name": "normal",
          "pokemon": Array [
            "16",
            "17",
          ],
        },
        Object {
          "id": "2",
          "name": "fighting",
          "pokemon": Array [
            "56",
            "57",
            "66",
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Card
            getPokemonInfo={[Function]}
            type={
              Object {
                "id": "1",
                "name": "normal",
                "pokemon": Array [
                  "16",
                  "17",
                ],
              }
            }
          />,
          <Card
            getPokemonInfo={[Function]}
            type={
              Object {
                "id": "2",
                "name": "fighting",
                "pokemon": Array [
                  "56",
                  "57",
                  "66",
                ],
              }
            }
          />,
        ],
        <Lists
          pokemons={undefined}
        />,
      ],
      "className": "PokeDeckContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "getPokemonInfo": [Function],
          "type": Object {
            "id": "1",
            "name": "normal",
            "pokemon": Array [
              "16",
              "17",
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "getPokemonInfo": [Function],
          "type": Object {
            "id": "2",
            "name": "fighting",
            "pokemon": Array [
              "56",
              "57",
              "66",
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "pokemons": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Card
              getPokemonInfo={[Function]}
              type={
                Object {
                  "id": "1",
                  "name": "normal",
                  "pokemon": Array [
                    "16",
                    "17",
                  ],
                }
              }
            />,
            <Card
              getPokemonInfo={[Function]}
              type={
                Object {
                  "id": "2",
                  "name": "fighting",
                  "pokemon": Array [
                    "56",
                    "57",
                    "66",
                  ],
                }
              }
            />,
          ],
          <Lists
            pokemons={undefined}
          />,
        ],
        "className": "PokeDeckContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "getPokemonInfo": [Function],
            "type": Object {
              "id": "1",
              "name": "normal",
              "pokemon": Array [
                "16",
                "17",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "getPokemonInfo": [Function],
            "type": Object {
              "id": "2",
              "name": "fighting",
              "pokemon": Array [
                "56",
                "57",
                "66",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "pokemons": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
